tests/mr_slice.rs

```diff
@@ -157,103 +151,94 @@ mod remote_mr_slice {
         assert_eq!(s3.addr(), rmr.addr() + s3_start);
         assert_eq!(s4.length(), 1);
         assert_eq!(s4.addr(), rmr.addr() + s4_pos);
-        sleep(Duration::from_secs(1)).await;
+        // wait for the agent thread to send all reponses to the remote.
+        tokio::time::sleep(Duration::from_secs(1)).await;
     }
 
-    #[tokio::test]
+    // #[tokio::test] has a default single-threaded test runtime that may cause a dead lock.
+    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
     async fn test() {
-        let addr = "127.0.0.1:19200";
+        let addr = SocketAddrV4::new(Ipv4Addr::new(127, 0, 0, 1), pick_unused_port().unwrap());
```

Comment:
This function pattern used in many tests. I recommend to wrap a function for it.